### PR TITLE
Adding osqp-eigen recipe

### LINKED ISSUE #15080: Package osqp-eigen
[`osqp-eigen`](https://github.com/robotology/osqp-eigen) is a simple C++ wrapper for [`osqp`](https://osqp.org/) library. It depends only on `osqp` and `Eigen3`, both of them are already available in conda-forge https://github.com/conda-forge/libosqp-feedstock and https://github.com/conda-forge/eigen-feedstock

Currently, it is already possible to install `osqp-eigen` using conda via [`robotogy`](https://anaconda.org/robotology/osqp-eigen) channel 
```
conda install -c robotology osqp-eigen
```
I think it would be nice to move `osqp-eigen` from `robotology` to `conda-forge` 

cc @traversaro

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

The PR is ready to be reviewed @conda-forge/help-c-cpp

This PR closes #15080 
cc @traversaro

### CODE REVIEW

File: recipes/osqp-eigen/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/robotology/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 3eb7011c8c706142a00c2679e9af5172ea504d56d0827cf3b519f2b78feee943
+
+build:
+  number: 1
```
Comment (traversaro) [inline recipes/osqp-eigen/meta.yaml:13]: 0

Comment (GiulioRomualdi) [reply recipes/osqp-eigen/meta.yaml]: Fixed here: https://github.com/conda-forge/staged-recipes/pull/15309/commits/8740f5142c813e1800a404f9683d2797a275ce6d

**Comment by traversaro** [comment]
@GiulioRomualdi feel free to add me as well as maintainer, it is a good practice to avoid having just one maintainer in recipes to reduce the bus factor.

**Comment by traversaro** [comment]
> The PR is ready to be reviewed @conda-forge/help-c-cpp

The CI is now green, @conda-forge/help-c-cpp this is now ready for review.

File: recipes/osqp-eigen/build.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+mkdir build && cd build
+
+cmake ${CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
```
Comment (traversaro) [inline recipes/osqp-eigen/build.sh:6]:
```suggestion
cmake ${CMAKE_ARGS} \
```
See https://github.com/conda-forge/staged-recipes/pull/15301#discussion_r652075460 and https://github.com/conda-forge/staged-recipes/pull/15140#discussion_r646172798 .

**Review by traversaro — CHANGES_REQUESTED** [review]

**Comment by traversaro** [comment]
@wolfv just FYI, this is the first package that we migrate from the `robotology` channel to `conda-forge`, hopefully many will follow in the future. : )

**Comment by wolfv** [comment]
Awesome!

**Pull request merged**